testing/test_minibatch_sequential_pipeline.py

```diff
@@ -96,3 +96,41 @@ def test_mini_batch_sequential_pipeline_should_fit_transform_steps_sequentially_
     assert tape4_fit.data == [([0, 8, 16, 24, 32, 40, 48, 56, 64, 72], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
                               ([80, 88, 96, 104, 112, 120, 128, 136, 144, 152], [10, 11, 12, 13, 14, 15, 16, 17, 18, 19])]
     assert tape4.name_tape == ["4", "4"]
+
+def test_minibatch_sequential_pipeline_change_batch_size_works():
+    tape1 = TapeCallbackFunction()
+    tape1_fit = TapeCallbackFunction()
+    tape2 = TapeCallbackFunction()
+    tape2_fit = TapeCallbackFunction()
+
+    p = MiniBatchSequentialPipeline([
+        MultiplyBy2FitTransformCallbackStep(tape1, tape1_fit, ["1"]),
+        Joiner(batch_size=10),
+        MultiplyBy2FitTransformCallbackStep(tape2, tape2_fit, ["2"]),
+        Joiner(batch_size=10)
+    ])
+
+    # When
+    p, outputs = p.fit_transform(list(range(20)), list(range(20)))
+    p.set_batch_size(5)
+    p, outputs = p.fit_transform(list(range(20,30)), list(range(20,30)))
+
+    # Then
+
+    assert tape1.data == [[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+                          [20, 21, 22, 23, 24], [25, 26, 27, 28, 29]]
+    assert tape1_fit.data == [([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+                              ([10, 11, 12, 13, 14, 15, 16, 17, 18, 19], [10, 11, 12, 13, 14, 15, 16, 17, 18, 19]),
```

Comment:
I know the data here was already defined above when you added this test, but in the future let's use smaller quantities of data in tests to make the code more legible. E.g.: a batch size of 3 instead of 10 would have been ok.

Reply:
deal.